src/Env.ts

```diff
@@ -0,0 +1,186 @@
+/**
+ * The Env comonad (aka the Coreader, Environment, or Product comonad)
+ *
+ * What the Store comonad is to the State monad, the Env comonad is to the
+ * Reader monad.
+ *
+ * ```ts
+ * import * as E from 'fp-ts/Env'
+ *
+ * const env: E.Env<number,string> = () => [5, "horse"]
+ *
+ * const repeat = (env: E.Env<number,string>):string => {
+ *   const numRepeat = E.ask(env)
+ *   const str = E.extract(env)
+ *   return str.repeat(numRepeat)
+ * }
+ *
+ * repeat(env) // "horsehorsehorsehorsehorse"
+ * ```
+ *
+ * @since 3.0.0
+ */
+import { Comonad2 } from './Comonad'
+import { identity, pipe, flow } from './function'
+import { Functor2 } from './Functor'
+import { Extend2 } from './Extend'
+
+// -------------------------------------------------------------------------------------
+// model
+// -------------------------------------------------------------------------------------
+
+// tslint:disable:readonly-array
+/**
+ * @category model
+ * @since 3.0.0
+ */
+export interface Env<E, A> {
+  (): [E, A]
+}
+// tslint:enable:readonly-array
+
+/**
+ * Returns the current environment
+ *
+ * @since 3.0.0
```

Comment:
I'm not sure what version to put here, so guessed an arbitrary one, happy to update.